lib/runtime/procs/join.js

```diff
@@ -801,11 +802,20 @@ class join extends base {
         return joinpts;
     }
     emit_result(points) {
-        // emit the points, possibly bracketed by marks, and update emitter state.
+        // Emit the points, possibly bracketed by marks, and update emitter state.
+        // Mark interval will be the distance between group start and complete times.
         var leader = this.batched_leader() ;
         var join_time = this.outer ? this.ins[this.outer - 1].live_time() : this.max_live_time();
+        var interval;
+
         if (leader) {
-            this.maybe_emit_mark(leader.live().start_time);
+            var live_leader = leader.live();
+            interval = live_leader.interval;
+            interval = interval || live_leader.complete_time.subtract(live_leader.start_time);
```

Comment:
When exactly does the situation that we don’t have `live_leader.interval` set arise?